sql-plugin/src/main/scala/com/nvidia/spark/rapids/SpillableColumnarBatch.scala

```diff
@@ -245,18 +338,29 @@ object SpillableColumnarBatch {
    */
   def apply(batch: ColumnarBatch,
       priority: Long): SpillableColumnarBatch = {
+    Cuda.DEFAULT_STREAM.sync()
```

Comment:
why?

Reply:
I'll add comments in the code around why we have the sync.

The reason is that if you hand off an object to the framework, we could turn around and spill them immediately. We need to make sure the object is immutable when it hits the store. 

We can move the sync back to the factory methods in the spill framework if that is desired?